qml-ts-mode.el

```diff
@@ -181,138 +183,262 @@ See `treesit-thing-settings' for more information."))
   "QML keywords for tree-sitter font-locking.")
 
 
-(defvar qml--treesit-font-lock-settings
-  (treesit-font-lock-rules
+(defun qml--is-node-id? (node)
+  "Check if NODE is an ui_binding node corresponding to the id."
+  (string= (treesit-node-text node) "id"))
 
-   :language 'qmljs
-   :feature 'comment
-   '((comment) @font-lock-comment-face)
 
-   :language 'qmljs
-   :feature 'constant
-   '(((identifier) @font-lock-constant-face
-      (:match "^[A-Z_][A-Z_\\d]*$" @font-lock-constant-face))
-     [(true) (false) (null)] @font-lock-constant-face)
+(defconst qml--treesit-font-lock-comment-settings
+  '((comment) @font-lock-comment-face))
 
-   :language 'qmljs
-   :feature 'keyword
-   `([,@js--treesit-keywords] @font-lock-keyword-face
-     [(this) (super)] @font-lock-keyword-face
-     [,@qml--treesit-keywords] @font-lock-keyword-face)
 
-   :language 'qmljs
-   :feature 'string
-   '((regex pattern: (regex_pattern)) @font-lock-regexp-face
-     (string) @font-lock-string-face)
+(defconst qml--treesit-font-lock-definitions-settings
+  '((function_declaration
+     name: (identifier) @font-lock-function-name-face)
 
-   :language 'qmljs
-   :feature 'string-interpolation
-   :override t
-   '((template_string) @js--fontify-template-string
-     (template_substitution ["${" "}"] @font-lock-misc-punctuation-face))
+    (variable_declarator
+     name: (identifier) @font-lock-variable-name-face)
 
-   :language 'qmljs
-   :feature 'type
-   '((type_identifier) @font-lock-type-face
+    (ui_property
+     name: (identifier) @font-lock-property-name-face)
 
-     (predefined_type) @font-lock-type-face
+    ;;   (variable_declarator
+    ;;    name: (array_pattern
+    ;;           (identifier)
+    ;;           (identifier)
+    ;;           @font-lock-function-name-face)
+    ;;    value: (array (number) (function)))
```

Comment:
This is part of the code that was broken with the current grammar. I'm not sure which QML code results in an `array_pattern` in the AST. We can either remove this, or fix it if there is something missing in the font-lock.